### PR TITLE
[IE VPU] Execution graph via ngraph for VPU plugin

### PR DESCRIPTION
**Pull request by adudchen**
#-33036

### CODE REVIEW

File: inference-engine/src/vpu/graph_transformer/src/utils/runtime_graph.cpp [hunk 1]
```diff
@@ -4,8 +4,12 @@
 
 #include "vpu/utils/runtime_graph.hpp"
 
+#include "ie_ngraph_utils.hpp"
```
Comment (DariaMityagina) [inline inference-engine/src/vpu/graph_transformer/src/utils/runtime_graph.cpp:7]: Not really important but `ie_ngraph_utils.hpp` includes `string`, so maybe line 16 is redundant now.

File: inference-engine/tests/functional/plugin/shared/include/behavior/exec_graph_info.hpp [hunk 2]
```diff
@@ -228,7 +228,7 @@ TEST_P(ExecGraphTests, CheckExecGraphInfoSerialization) {
     // Create CNNNetwork from ngrpah::Function
     InferenceEngine::CNNNetwork cnnNet(function);
     InferenceEngine::CNNNetwork execGraph;
-    if (targetDevice == CommonTestUtils::DEVICE_CPU || targetDevice == CommonTestUtils::DEVICE_GPU) {
+    if (targetDevice == CommonTestUtils::DEVICE_CPU || targetDevice == CommonTestUtils::DEVICE_GPU || targetDevice == CommonTestUtils::DEVICE_MYRIAD) {
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/shared/include/behavior/exec_graph_info.hpp:231]: could you please just remote this `if`? the body of `if` should be the same for all devices.
BTW, don't we need to enable this tests for HDDL as well?

File: inference-engine/src/vpu/graph_transformer/src/utils/runtime_graph.cpp [hunk 3]
```diff
@@ -4,8 +4,12 @@
...
+#include "inference_engine.hpp"
```
Comment (ilya-lavrenov) [inline inference-engine/src/vpu/graph_transformer/src/utils/runtime_graph.cpp:8]: why do we need `inference_engine.hpp`?

**Comment by ilya-lavrenov** [comment]
@adudchen let's finalize by 2021.1

File: inference-engine/tests/functional/plugin/shared/include/behavior/exec_graph_info.hpp (hunk 2 above)
Comment (adudchen) [reply inference-engine/tests/functional/plugin/shared/include/behavior/exec_graph_info.hpp]: I'll enable these tests for the HDDL plugin after merging this PR.

**Comment by ilya-lavrenov** [comment]
@adudchen please, merge with latest master

**Comment by ilya-lavrenov** [comment]
@adudchen please, have a look https://openvino-ci.intel.com/job/private-ci/job/ie/job/ie-tests-linux-ubuntu18-gna/12296/

File: inference-engine/src/vpu/graph_transformer/src/utils/runtime_graph.cpp [hunk 4]
```diff
@@ -17,49 +21,117 @@ using namespace InferenceEngine;
...
-        std::map<size_t, CNNLayerPtr> stageMetaIndexToLayer;
+std::map<std::string, std::string> extractMeta(const StageMetaInfo&);
 
-        auto createLayerFromMeta = [&](const StageMetaInfo &stageMetaInfo) -> CNNLayer::Ptr {
-            auto layer = std::make_shared<CNNLayer>(LayerParams{stageMetaInfo.stageName,
-                                                                stageMetaInfo.layerType,
-                                                                Precision::FP16});
+}  // namespace
+
+    InferenceEngine::ICNNNetwork::Ptr buildRuntimeGraphAsIeNgraphNet(GraphMetaInfo &graphMetaInfo, const std::vector<float>& perfInfo) {
```
[the remark this answers is not in the document]
Comment (adudchen) [reply inference-engine/src/vpu/graph_transformer/src/utils/runtime_graph.cpp]: Fixed

**Review by Maxim-Doronin — APPROVED** [review]
Some style issues. But in general LGTM

**Comment by Maxim-Doronin** [comment]
@ilya-lavrenov, can we merge this PR right now?

**Comment by ilya-lavrenov** [comment]
@Maxim-Doronin I think yes, since it's passed review from VPU maintainers

**Pull request merged**